rest-api-spec/src/main/resources/rest-api-spec/api/delete_by_query_rethrottle.json

```diff
@@ -0,0 +1,25 @@
+{
+  "delete_by_query_rethrottle": {
```

Comment:
I didn't really want to copy the existing "reindex_rethrottle.json" twice, expecially since the "paths" contain all the endpoint variants, but RestHighLevelClientTests#testApiNamingConventions complained to I did it. Not sure if this is necessary or if there are other ways to satisfy the check.

Reply:
Maybe drop the variant paths and keep the new files? That seems most in the spirit of "these are three different things that *happen* to be the same thing right now."

Reply:
I added an update